client/src/app/page/create-update-user-page/create-update-side-bar-info/create-update-side-bar-info.component.ts

```diff
@@ -64,18 +62,18 @@ export class CreateUpdateSideBarInfoComponent implements OnInit, OnDestroy {
       });
   }
 
-  ngOnDestroy(): void {
-    this.destroy$.next();
-    this.destroy$.complete();
-  }
-
   selectDepartment(id: any): void {
     this.user.department = id;
     this.ifChosenDevelopmentDepartment = id === '5cab28b4e5773a19a4462fd1';
     this.ifChosenHrDepartment = id === '5cb9c437b5cfd134acc5783e';
     this.positions = this.departments
-      .filter(elem => elem._id === id)[0].position
+      .find(elem => elem._id === id).position
       .map(e => new OptionPair(e, e));
     this.userService.chosenDepartment.emit();
   }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
```

Comment:
do you really need next here?

Reply:
this unsubscribes from the observable, isn't it?